openrr-apps/src/utils.rs

```diff
@@ -126,7 +137,119 @@ pub fn init_with_anonymize(name: &str, config: &RobotConfig) {
         .with_writer(io::stderr)
         .init();
 }
+
+#[derive(Clone)]
+pub struct OpenrrFormatter {
+    formatter: Format,
+    name: String,
+}
+
+impl OpenrrFormatter {
+    fn new(name: String) -> Self {
+        Self {
+            formatter: tracing_subscriber::fmt::format(),
+            name,
+        }
+    }
+}
+
+impl<S, N> FormatEvent<S, N> for OpenrrFormatter
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+    N: for<'a> FormatFields<'a> + 'static,
+{
+    fn format_event(
+        &self,
+        ctx: &FmtContext<'_, S, N>,
+        writer: &mut dyn std::fmt::Write,
+        event: &Event<'_>,
+    ) -> std::fmt::Result {
+        write!(writer, "{} ", self.name)?;
+        self.formatter.format_event(ctx, writer, event)
+    }
+}
+
+pub fn init_tracing_with_file_appender(config: LogConfig, name: String) -> WorkerGuard {
+    let default_level = match config.level {
+        LogLevel::TRACE => Level::TRACE,
+        LogLevel::DEBUG => Level::DEBUG,
+        LogLevel::INFO => Level::INFO,
+        LogLevel::WARN => Level::WARN,
+        LogLevel::ERROR => Level::ERROR,
+    };
+    let rotation = match config.rotation {
+        LogRotation::MINUTELY => Rotation::MINUTELY,
+        LogRotation::HOURLY => Rotation::HOURLY,
+        LogRotation::DAILY => Rotation::DAILY,
+        LogRotation::NEVER => Rotation::NEVER,
+    };
+    let formatter = OpenrrFormatter::new(name);
+    let file_appender = RollingFileAppender::new(rotation, config.directory, config.prefix);
+    let (file_writer, guard) = tracing_appender::non_blocking(file_appender);
+    let subscriber = tracing_subscriber::registry()
+        .with(EnvFilter::from_default_env().add_directive(default_level.into()))
+        .with(
+            Layer::new()
+                .event_format(formatter.clone())
+                .with_writer(io::stderr),
+        )
+        .with(
+            Layer::new()
+                .event_format(formatter)
+                .with_writer(file_writer),
+        );
+    tracing::subscriber::set_global_default(subscriber).expect("Unable to set a global collector");
+    guard
+}
+
+#[derive(Deserialize)]
+pub enum LogLevel {
+    TRACE,
+    DEBUG,
+    INFO,
+    WARN,
+    ERROR,
+}
+impl Default for LogLevel {
+    fn default() -> Self {
+        Self::INFO
+    }
+}
+
+#[derive(Deserialize)]
+pub enum LogRotation {
+    MINUTELY,
+    HOURLY,
+    DAILY,
+    NEVER,
+}
+impl Default for LogRotation {
+    fn default() -> Self {
+        Self::HOURLY
+    }
+}
+
+#[derive(Deserialize, Default)]
+#[serde(default)]
+pub struct LogConfig {
+    #[serde(default)]
+    pub level: LogLevel,
+    #[serde(default)]
+    pub rotation: LogRotation,
+    #[serde(default = "default_log_directory")]
+    pub directory: String,
+    #[serde(default = "default_log_prefix")]
+    pub prefix: String,
+}
+
+fn default_log_prefix() -> String {
+    "log".to_string()
+}
+
+fn default_log_directory() -> String {
+    "/tmp".to_string()
```

Comment:
I think `/tmp` does not work on windows.

How about using rust function?
https://doc.rust-lang.org/std/env/fn.temp_dir.html